### DISCUSSION TITLE
How to style disabled input in theme.js ?

### DISCUSSION THREAD

**Discussion by ngprnk** [chakra-ui/chakra-ui#4332] [Q&A]
I want to set the opacity of disabled input to 1 is this the correct way to do it ? 

```javascript
const Input =  {
  baseStyle: {
    "_disabled": {
      opacity: 1,
    }
  }
}
```
```javascript
const theme = extendTheme({
  components: {
    Input,
  },
});

```

**Comment by chasinhues** [comment]
Check out the source code for the input theme: https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/input.ts

You'll notice that the componet has [`parts`](https://github.com/chakra-ui/chakra-ui/blob/1890fb4fc48d784ad173b9827d5f27c503d0ed54/packages/theme/src/components/input.ts#L3), meaning it's a multipart component ([read more here](https://chakra-ui.com/docs/theming/component-style#single-part-and-multipart-components)).

So what you actually need to do is:

```jsx
const Input =  {
  baseStyle: {
    // We apply the styles to a part of the component.
    // In this case, the "field" part.
    field: {
      "_disabled": {
        opacity: 1,
      }
    }
  }
}
```

**Reply by ngprnk** [reply]
Hey @chasinhues  thanks for the reply. The code below solves my problem.

```javascript
const  Input =  {
      variants: {
        outline: {
          field: {
            _disabled: {
              opacity: 1,
            }
          }
        }
      }
    }

```
Here is a really good reference of codesandbox. 
https://codesandbox.io/s/chakra-ui-theme-extension-w5u2n
